CMake/resolve_dependency_modules/fmt.cmake

```diff
@@ -19,9 +19,9 @@ else()
   set(VELOX_FMT_VERSION 8.0.1)
   set(VELOX_FMT_SOURCE_URL
       "https://github.com/fmtlib/fmt/archive/${VELOX_FMT_VERSION}.tar.gz")
-  set(VELOX_FMT_BUILD_SHA256_CHECKSUM
-      b06ca3130158c625848f3fb7418f235155a4d389b2abc3a6245fb01cb0eb1e01)
 endif()
+set(VELOX_FMT_BUILD_SHA256_CHECKSUM
```

Comment:
Not sure I understand this, the sha is linked to version and source url, we shouldnt set this if FMT_URL is set.

Reply:
Sure, sha should bound with version. 
I'm not sure Velox dependency management want user must use specific dependency version to build or allow user use different version, if first is right, we need to keep version and sha whether FMT_URL is set or not, if second, we need provide another FetchContent_Declare let user pass sha as FMT_SHA.

I prefer keep specific version and sha, what community's suggestions? @kgpai @assignUser